src/overlays/actors/ovl_Obj_HsStump/z_obj_hsstump.c

```diff
@@ -23,18 +28,132 @@ const ActorInit Obj_HsStump_InitVars = {
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_060011B0);
+    if (this->isHidden != 0) {
+        if (this->isHidden != 1) {
+            return;
+        }
+        if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+            this->isHidden = 0;
+        } else {
+            this->dyna.actor.draw = NULL;
+            Actor_SetScale(&this->dyna.actor, 0.0f);
+            func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        }
+    }
+    ObjHsStump_SetupIdle(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_HsStump_0x80BDAA30/func_80BDAB6C.asm")
+void ObjHsStump_SetupIdle(ObjHsStump* this, GlobalContext* globalCtx) {
+    this->actionFunc = ObjHsStump_Idle;
+}
 
+void ObjHsStump_Idle(ObjHsStump* this, GlobalContext* globalCtx) {
+    if (this->isHidden == 1) {
+        if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+            ObjHsStump_SetupAppear(this, globalCtx);
+        }
+    }
+}
+
+void ObjHsStump_SetupAppear(ObjHsStump* this, GlobalContext* globalCtx) {
+    this->dyna.actor.draw = ObjHsStump_Draw;
+    this->framesAppeared = 0;
+    this->rotAngle = 0;
+    this->rotFactor = 3640.0f;
+    func_8019F128(0x3A86);
+    this->actionFunc = func_80BDABCC;
+}
+
+#ifdef NON_MATCHING
+// Correct instructions, but they are all out of order in the (this->framesAppeared) < 11 branch
+void func_80BDABCC(ObjHsStump* this, GlobalContext* globalCtx) {
+    s32 i;
+    s32 angle;
+    s16 numDirections;
+    Vec3f iceSmokePosOffset;
+    Vec3f iceSmokeVelOffset;
+    s16 offsetAngle;
+    Vec3f iceSmokeVel;
+    f32 angleBAMS;
+    Vec3f iceSmokePos;
+
+    if (this->framesAppeared >= 0) {
+        Math_SmoothStepToF(&this->rotFactor, 0.0f, 1.0f, this->framesAppeared + 18.0f, 0.01f);
+        this->dyna.actor.shape.rot.x = (Math_SinS(this->rotAngle) * this->rotFactor) + this->dyna.actor.home.rot.x;
+        this->dyna.actor.shape.rot.z = (Math_SinS(this->rotAngle * 2) * this->rotFactor) + this->dyna.actor.home.rot.z;
+        this->rotAngle += 0x2000;
+    }
+    if (this->framesAppeared < 11) {
+        if (this->framesAppeared == 0) {
+            iceSmokePosOffset.x = 1.0f;
+            iceSmokePosOffset.y = 0.5f;
+            iceSmokePosOffset.z = 0.0f;
+            iceSmokeVelOffset.x = 1.0f;
+            iceSmokeVelOffset.y = 0.5f;
+            iceSmokeVelOffset.z = 0.0f;
+
+            numDirections = 4;
+            angle = 360.0f / numDirections;
+            angleBAMS = angle * (0x10000 / 360.0f);
+
+            for (i = 0; i < numDirections; i++) {
+                offsetAngle = i * angleBAMS;
+                Lib_Vec3f_TranslateAndRotateY(&this->dyna.actor.world.pos, offsetAngle, &iceSmokePosOffset,
+                                              &iceSmokePos);
+                Lib_Vec3f_TranslateAndRotateY(&D_801D15B0, offsetAngle, &iceSmokeVelOffset, &iceSmokeVel);
+                EffectSsIceSmoke_Spawn(globalCtx, &iceSmokePos, &iceSmokeVel, &D_80BDAF84, 0x64);
```

Comment:
You could probably replace 0x64 with 100

Reply:
👍